server/src/main/java/org/elasticsearch/common/util/concurrent/ThreadContext.java

```diff
@@ -798,6 +799,24 @@ private ThreadContextStruct putResponseHeaders(Map<String, Set<String>> headers)
             return new ThreadContextStruct(requestHeaders, newResponseHeaders, transientHeaders, isSystemContext);
         }
 
+        private void logWarningHeaderThresholdExceeded(long threshold, Setting<?> thresholdSetting) {
+            // If available, log some selected headers to help identifying the source of the request.
+            // Note: Only Task.HEADERS_TO_COPY are guaranteed to be preserved at this point.
```

Comment:
TBD: From a user's perspective it would be convenient to also include `User-Agent` in here. Though, currently, that isn't part of the headers that are copied over. I'm on the fence if to add it or not.

Reply:
I'd be inclined not to. We don't use this to identify requests elsewhere.